tests/org.eclipse.swt.tests.win32/JUnit Tests/org/eclipse/swt/widgets/WidgetWin32Tests.java

```diff
@@ -0,0 +1,53 @@
+package org.eclipse.swt.widgets;
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.internal.DPITestUtil;
+import org.eclipse.swt.internal.DPIUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class WidgetWin32Tests {
+	private Display display;
+	private boolean autoScaleOnRuntime;
+
+	@Before
+	public void setUp() {
+		autoScaleOnRuntime = DPIUtil.isAutoScaleOnRuntimeActive();
+		DPITestUtil.setAutoScaleOnRunTime(true);
+		display = Display.getDefault();
+	}
+
+	@After
+	public void tearDown() {
+		DPITestUtil.setAutoScaleOnRunTime(autoScaleOnRuntime);
+	}
+
+	@Test
+	public void widgetZoomShouldChangeOnZoomLevelChange() {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+		Shell shell = new Shell(display);
+
+		Button button = new Button(shell, SWT.PUSH);
+		button.setBounds(0, 0, 200, 50);
+		button.setText("Widget Test");
+		button.setBackground(display.getSystemColor(SWT.COLOR_CYAN));
+		shell.open();
+		changeZoomLevel(scaledZoom, shell);
+		assertEquals("The Zoom Level should be updated for button on zoom change event on its shell", scaledZoom, button.getZoom());
+	}
+
+	public void changeZoomLevel(int zoomLevel, Control widget) {
```

Comment:
Does this one really need to be `public`?

Reply:
nope. removed the whole method since it is not needed anymore. Extending to the base test for the method